platform/posix/transport/src/plaintext_posix.c

```diff
@@ -201,7 +205,10 @@ int32_t Plaintext_Send( NetworkContext_t * pNetworkContext,
     /* Set the file descriptor for poll. */
     pollFds.fd = pPlaintextParams->socketDescriptor;
 
-    /* Check if there is data to read from the socket. */
+    /* Check if data can be written to the socket.
+     * Note: This is done to avoid blocking on send() call when
+     * the socket is ready to accept more data for transmitting
+     * (possibly due to a full TX buffer). */
```

Comment:
```suggestion
     * Note: This is done to avoid blocking on send() call when
     * the socket is not ready to accept more data for networl
     * transmission (possibly due to a full TX buffer). */
```